### ISSUE TITLE
Contravariance Weirdness

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#1784]
**Describe the bug**
Contravariance bug. I think the code + examples best describe it as not sure exactly what's off. Also error message becomes less informative when contravariance is true.
 
**To Reproduce**

```python
from typing import Any, TypeVar

from typing_extensions import Protocol

InputT = TypeVar("InputT", contravariant=True)
OutputT = TypeVar("OutputT", covariant=True)


class MyCallable1(Protocol[InputT]):
    def __call__(self, inputs: InputT) -> int:
        ...


class MyCallable2(Protocol[InputT, OutputT]):
    def __call__(self, inputs: InputT) -> OutputT:
        ...


class Foo:
    def __call__(self, inputs) -> int:
        return 5


x = Foo()
h: MyCallable1[int] = x  # OK
g: MyCallable2[int, int] = x  # Error on input type.

```

Type checking that I would expect no errors. I get an error on the last line with the error message, 

```
Expression of type "Foo" cannot be assigned to declared type "MyCallable2[int, int]"
  TypeVar "InputT@MyCallable2" is contravariant
    Type "int" cannot be assigned to type "InputT@MyCallable2"

```

Beyond error not making sense to me on how h type checks but g fails even though g's complaint is on input type and not added output type, the other issue is the error message has a confusing description of the type being InputT@MyCallable2. When I drop contravariance I get an error message of,

```
Expression of type "Foo" cannot be assigned to declared type "MyCallable2[int, int]"
  TypeVar "InputT@MyCallable2" is invariant
    "object" is incompatible with "int"
```
This time error makes more sense. A bit confused why object instead of Any, but otherwise sure. So why does object become opaque  "InputT@MyCallable2"?

Other weird thing playing with this is if I explicitly add "object" as a type to inputs in Foo the error goes away. Adding explicit Any keeps the error though. Shouldn't Any be treated like both a subtype/supertype of everything?

**VS Code extension or command-line**
pyright 1.1.127.

**Comment by erictraut** [comment]
Thanks for the bug report. I've fixed the underlying cause of the bug and added your example to our unit tests to prevent future regressions.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.134, which I just released.
